source/dem/particle_wall_contact_force.cc

```diff
@@ -158,209 +165,445 @@ ParticleWallContactForce<dim,
-                            normal_force,
-                            tangential_force,
-                            tangential_torque,
-                            rolling_resistance_torque,
-                            particle_torque,
-                            particle_force);
+                          contact_record[particle->get_local_index()]
+                            .emplace_back(triangle_cell_iterator,
+                                          normal_overlap,
+                                          contact_indicators[particle_counter],
+                                          contact_info);
                         }
-                      else
+                    }
+                  particle_counter++;
+                }
+            }
+        }
+      // Every contact between this solid object and the particles are
+      // recorded. Now, we need to remove the invalid contacts from
+      // the records and compute the forces associated with the valid
+      // contacts.
+
+      // Loop on every particle / contact record
+      for (auto &[particle_index, this_contact_record] : contact_record)
+        {
+          // Loop on every contact in the contact record
+          for (auto contact_1 = this_contact_record.begin();
+               contact_1 != this_contact_record.end();)
+            {
+              // Extract the information of C1
+              auto &[T1_cell,
+                     normal_overlap_C1,
+                     contact_indicator_C1,
+                     contact_info_C1] = *contact_1;
+
+              // Assigning the triangle neighboring list of T1;
+              const std::vector<
+                typename Triangulation<dim - 1, dim>::active_cell_iterator>
+                &T1_es_neighbors = this_solid_es_neighbors.at(T1_cell);
+              const std::vector<
+                typename Triangulation<dim - 1, dim>::active_cell_iterator>
+                &T1_vs_neighbors = this_solid_vs_neighbors.at(T1_cell);
+
+              // Initialize variables. We need to compare C1 to every other
+              // contact that was compared yet.
+              bool erase_contact_1 = false;
+              auto contact_2       = std::next(contact_1);
+              while (contact_2 != this_contact_record.end())
+                {
+                  // Extract the information of C2
+                  auto &[T2_cell,
+                         normal_overlap_C2,
+                         contact_indicator_C2,
+                         contact_info_C2] = *contact_2;
+
+                  // First, we check if both triangle are neighbors. If they
+                  // are not neighbors, C1 and C2 are automatically valid.
+                  // (Disconnected triangles)
+                  if (std::ranges::find(T1_es_neighbors, T2_cell) ==
+                        T1_es_neighbors.end() &&
+                      std::ranges::find(T1_es_neighbors, T2_cell) ==
+                        T1_es_neighbors.end())
```

Comment:
Isn't this twice the same check?
```suggestion
                  if (std::ranges::find(T1_es_neighbors, T2_cell) ==
                        T1_es_neighbors.end())
```